### DISCUSSION TITLE
How to integrate with MailHog in order to check emails?

### DISCUSSION THREAD

**Discussion by MagePsycho** [wardenenv/warden#565] [Help & Support]
My warden environment is working fine. I can access installed Magento and MailHog (https://mailhog.warden.test/)
But when I place an order or register as a customer, I don't see any emails in MailHog.

Is there any configuration for MailHog with Warden?

**Comment by maximecarey** [comment]
If you look at MailHog documentation, https://github.com/mailhog/MailHog, you can see the SMTP service is listening by default on 1025. The port is also exposed through docker. So your container should be able to connect to the SMTP server to send email using mailhog:1025

**Comment by MagePsycho** [comment]
Does that mean, we need to install some SMTP extension and configure `host=mailhog` and `port=1025`?

**Comment by maximecarey** [comment]
Inside a warden shell, if you run a php -i | grep sendmail, you should see the default email configuration. mhsendmail, the mailhog sendmail replacement should be installed inside your container and pointing to your local mailhog instance. It should work out of the box.

Try running this:
```
<?php
$message = "This is a test!!";
$headers = "From: from@test";

mail("target@provider,test", "Testing", $message, $headers);
```

**Comment by navarr** [comment]
Following up on Maxime's message, the automatic configuration of PHP's sendmail should result in the `mail` command calls ending up in Mailhog.

Assuming that works, but Magento doesn't - there are a few different things that you should look for:

* Ensure you don't have an SMTP module or other type of mail module or capture module installed
* Ensure that you don't have email queuing turned on (or, run `bin/magento cron:run` manually a few times)

**Comment by MagePsycho** [comment]
@werfu @navarr grepping `sendmail` gives
```
sendmail_from => no value => no value
sendmail_path => /usr/local/bin/mhsendmail --smtp-addr='mailhog:1025' => /usr/local/bin/mhsendmail --smtp-addr='mailhog:1025'
Path to sendmail => /usr/local/bin/mhsendmail --smtp-addr='mailhog:1025'
```
But when I try to send via a custom script (script.php)
```
<?php
$message = "This is a test!!";
$headers = "From: from@test";
try {

	mail("rajendra.bhatta@citruss.com", "Testing", $message, $headers);
	echo 'Success';
} catch (Exception $e) {
	echo 'Exception:' . $e->getMessage();
}
```

And running `php -f script.php` results in
> error sending mail
> 2022/11/15 13:50:19 dial tcp 184.168.221.27:1025: getsockopt: connection timed out

**Comment by dudzio12** [comment]
@MagePsycho I've just tried your script on my env and it worked as expected.

```
sendmail_from => no value => no value
sendmail_path => /usr/local/bin/mhsendmail --smtp-addr='mailhog:1025' => /usr/local/bin/mhsendmail --smtp-addr='mailhog:1025'
Path to sendmail => /usr/local/bin/mhsendmail --smtp-addr='mailhog:1025'
```

**Comment by maximecarey** [comment]
@MagePsycho if you run a `docker ps` command, do you see a mailhog container running?

**Comment by davidalger** [comment]
Based on the reference to `184.168.221.27` the issue here is `mailhog` is being resolved to `mailhog.net` likely as a result of a search domain being present in `resolvconf` More on this here: https://github.com/davidalger/warden/issues/415#issuecomment-921825274

**Comment by christian-walker** [comment]
@MagePsycho / @werfu @navarr  were you guys ever able to get mailhog emails to work?

**Comment by bap14** [comment]
@christian-walker Out of the box, it should "just work". There are some prerequisites though:

1. You've started the warden service (e.g. `warden svc up`)
2. You're using a Warden environment (magento2, wordpress, etc.)
3. You're script is being executed from a running warden project

Are you running an actual Warden project, or are you trying to use just the Warden docker image itself in your own `docker-compose.yml` file?

**Reply by christian-walker** [reply]
@bap14 thanks! My issue was with a 3rd party module disallowing emails to go through in general.
